File: AdventOfCode.Domain/SantasFloor.fs
namespace AdventOfCode

module SantasFloor =

  open System
  open System.IO

  let loadFile =
    __SOURCE_DIRECTORY__ + "/inputs/floors.txt"
    |> File.ReadAllText

  let split = 
    List.ofSeq loadFile |> List.toArray

  let addFloor s =
    match s with
    | '(' -> 1
    | ')' -> -1
    | _ ->  0

  let floorCount = 
    split
    |> Array.map addFloor 
    |> Array.sum

  let positionBasement =
    split
    |> Array.map addFloor
    |> Array.scan (+) 0
    |> Array.findIndex(fun f -> f = -1)

File: AdventOfCode.CSharp/SantasFloor.cs
using System.Diagnostics.CodeAnalysis;
using System.IO;
using System.Linq;

namespace AdventOfCode.CSharp
{
  public class SantasFloor
  {

    public string FloorInstructions()
    {
      return File.ReadAllText(@"E:\labdays\AdventOfCode\AdventOfCode.CSharp\inputs\floors.txt");
    }

    public int EndFloor()
    {
      return FloorInstructions().Sum(f => MapFloor(f));
    }

    public int MapFloor(char c)
    {
      switch (c)
      {
        case '(':
          return 1;
        case ')':
          return -1;
        default:
          return 0;
      }
    }
  }
}